packages/language-server/src/lib/ast/configBlock.ts

```diff
@@ -32,15 +32,19 @@ export function getFirstDatasourceProvider(schema: PrismaSchema): string | undef
 }
 
 export function getAllPreviewFeaturesFromGenerators(schema: PrismaSchema): PreviewFeatures[] | undefined {
-  // matches any `previewFeatures = [x]` in any position
-  // thanks to https://regex101.com for the online scratchpad
-  const previewFeaturesRegex = /previewFeatures\s=\s(\[.*\])/g
+  /**
+   * ```prisma
+   * generator client {
+   *   provider        = "prisma-client-js"
+   *   // previewFeatures = [] // This will be ignored
+   *   previewFeatures = ["views"]
+   * }
+   * ```
+   *
+   * ? for more info: https://regex101.com/r/ezoTU2/1
+   */
+  const previewFeaturesRegex = /^\s*(?!\/\/\s)previewFeatures\s=\s(\[.*\])/m
```

Comment:
```suggestion
   * ```prisma
   * generator client {
   *   provider        = "prisma-client-js"
   *   // previewFeatures = [] // This will be ignored
   *   previewFeatures  =  ["views"]
   * }
   * ```
   *
   * ? for more info: https://regex101.com/r/ezoTU2/2
   */
  const previewFeaturesRegex = /^\s*(?!\/\/\s)previewFeatures\s*=\s*(\[.*\])/m
```

I just noticed it was too strict, it should allow multiple spaces on each side of the equal sign.